src/prefect/cli/flow_run.py

```diff
@@ -139,13 +139,25 @@ async def cancel(id: UUID):
 
 
 @flow_run_app.command()
-async def logs(id: UUID):
+async def logs(
+    id: UUID,
+    head: int = typer.Option(
+        None,
+        help="Number of lines to show from the head of the log",
+    ),
+):
     """
     View logs for a flow run.
     """
+    # Pagination - API returns max 200 lines at a time
     page_size = 200
     offset = 0
     more_logs = True
+
+    # If head is specified, we need to stop after we've retrieved enough lines
+    remaining_logs = head
+    num_logs_returned = 0
```

Comment:
I think `num_logs_returned` could go underneath `more_logs`, since it doesn't have to do with whether the head is specified.

Reply:
Sure!